crates/blockchain-tree/src/blockchain_tree.rs

```diff
@@ -911,10 +912,15 @@ impl<DB: Database, EF: ExecutorFactory> BlockchainTree<DB, EF> {
     #[track_caller]
     #[instrument(level = "trace", skip(self), target = "blockchain_tree")]
     pub fn make_canonical(&mut self, block_hash: &BlockHash) -> RethResult<CanonicalOutcome> {
+        let mut timings = Vec::new();
+
+        let start = Instant::now();
```

Comment:
I think a helper type that wraps an instant and tracks durations as fields would be more appropriate then putting messages onto a vec.

like

```rust
struct MakeCanonicalDurationsRecorder {
  now: Instant,
  
  durations: ...,
  /// this can be used to get rid of multiple Instant::now, since elapsed is monotonically nondecreasing
  latest: Option<Duration>
}


struct MakeCanonicalDurations {
  ...
}
```

and since

Reply:
just when I finished with helper types, I realized that's what tracing and spans were made for. Would be probably cleaner to use https://crates.io/crates/tracing-timing to measure inter-event timings.

Reply:
ok, I went for a similar approach but with our own recorder impl, so we can easily both report metrics and log timings while not resorting to passing closures everywhere.